__tests__/app.test.js

```diff
@@ -0,0 +1,304 @@
+            .send(noUserExistComment)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("no user exists")
+        })
+    })
+    test("404 - responds with error message when article id does not exist", () => {
+        return request(app)
+            .post(`/api/articles/99999/comments`)
+            .send(testComment)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("no comments in article found with ID 99999")
+        })
+    })
+})
+describe('PATCH /api/articles/:article_id', () => {
+    const testPatch = { inc_votes: 1 }
+    const testPatchMinus = { inc_votes: -100 }
+    const errorPatch = { inc_votes: "one" }
+    const articleId1 = {article_id: 1,
+                        title: "Living in the shadow of a great man",
+                        topic: "mitch",
+                        author: "butter_bridge",
+                        body: "I find this existence challenging",
+                        created_at: 1594329060000,
+                        votes: 100,
+                        article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+                    }
+    test("200 - responds with updated article and new plus vote count", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send(testPatch)
+            .expect(200)
+            .then(({ body }) => {
+                const updatedVotes = body.article.votes
+                const expectedVotes = articleId1.votes + testPatch.inc_votes
+                expect(updatedVotes).toBe(expectedVotes)
+        })
+    })
+    test("200 - responds with updated article and new minus vote count", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send(testPatchMinus)
+            .expect(200)
+            .then(({ body }) => {
+                const updatedVotes = body.article.votes
+                const expectedVotes = articleId1.votes + testPatchMinus.inc_votes
+                expect(updatedVotes).toBe(expectedVotes)
+        })
+    })
+    test("400 - responds with error message when patch is not a number", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send(errorPatch)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+        })
+    })
+    test("400 - responds with error message when article id is invalid", () => {
+        return request(app)
+            .patch("/api/articles/one")
+            .send(testPatch)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("bad request")
+        })
+    })
+    test("400 - responds with error message when patch is empty", () => {
+        return request(app)
+            .patch("/api/articles/1")
```

Comment:
I think the test as a whole is fine but the description could be better. This is not specifically covering if the body (would say body rather than patch) is empty, but in general if the not keys are found. We would expect the same behaviour if there was a body with keys like { roseiscool: true }